lib/Base/Tensor.cpp

```diff
@@ -379,6 +380,60 @@ std::string Tensor::toString(unsigned maxNumElem) const {
   return os.str();
 }
 
+void Tensor::dumpToRawBinaryFile(const char *filename) {
```

Comment:
I feel like dumping some meta-information about the type of the tensor could be a good idea. Otherwise there is a high-risk of misinterpreting the content of the dumped tensor later.

Reply:
Also, this approach currently supports dumping a single tensor per file if I understand correctly. But what if we want to dump e.g. all tensors used during the computation. There could be thousands of them. I wonder if it makes sense to support scenarios where multiple tensors could be dumped into/read from the same file.

Reply:
Nit: Maybe all these serialization related functions should be in a separate file like Serialization.cpp. 

And I'm not sure they need to be methods on Tensor, especially if we'd support dumping multiple tensors into the same file.

Reply:
See the comments above. The profiling application just needs the loading methods, one tensor per file, better without the meta info in order to be easier to be externally generated.
In this case, do you think I should move them in ModelProfiler.cpp?
I thought that maybe that simpler format would be reutilized in some other places, especially for the front-end tools.

Reply:
I think that the functionality for dumping/reading tensors is generally useful. Not only for profiling purposes, but e.g. for numerical testing purposes, etc.

But I'm still very worried about a format that does not include any meta-information (e.g. payload type and dimensions; maybe optional tensor name) as such a format is very likely to result in a lot of troubles if the mapping between tensors to be read and the dumped content files get mixed up. 

@kunmingho You actually looked into the same issues. Would you like to comment based on your experience with this?

Reply:
I would prefer these functions as-is. I enhanced the description of the methods (after all they contain the name "raw" in the method names which suggests no meta information). These methods are useful when the tensors are already created and part of the graph (loading model input tensors, dumping model output tensors).

For the mechanisms you mention I think other methods should be developed (much more complex). In order for the file to be sufficient for a tensor description, a lot of meta-information should be there: data type, shape, padding/strides, quantization parameters (scale/offset).